source/simulator.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+#include "scheduler.hpp"
+
+namespace sim {
+
+class Simulator {
+public:
+    Simulator();
+    void create_device(std::string a_name);
```

Comment:
I think it will be easy and usable to return created device^
```suggestion
    Device* create_device(std::string a_name);
```

Reply:
I do not think this is needed for now. I have renamed the functions to `add_device` and `add_link` to make it clear that we add a device/link to the network.